packages/routers/src/StackRouter.tsx

```diff
@@ -47,36 +48,69 @@ export default function StackRouter(options: StackRouterOptions) {
     ...BaseRouter,
 
     getInitialState({ routeNames, routeParamList }) {
-      const index =
-        options.initialRouteName === undefined
-          ? 0
-          : routeNames.indexOf(options.initialRouteName);
+      const initialRouteName =
+        options.initialRouteName !== undefined
+          ? options.initialRouteName
+          : routeNames[0];
 
       return {
         key: `stack-${shortid()}`,
-        index,
+        index: 0,
         routeNames,
-        routes: routeNames.slice(0, index + 1).map(name => ({
-          name,
-          key: `${name}-${shortid()}`,
-          params: routeParamList[name],
-        })),
+        routes: [
```

Comment:
makes sense, but AFAIK it's unrelated? Can you confirm?

Reply:
yeah, not strictly related